pkg/genericapiserver/storage_factory.go

```diff
@@ -88,6 +88,11 @@ type groupResourceOverrides struct {
 	// of exposing one set of concepts.  autoscaling.HPA and extensions.HPA as a for instance
 	// The order of the slice matters!  It is the priority order of lookup for finding a storage location
 	cohabitatingResources []unversioned.GroupResource
+
+	// ignoreCohabitatingStorageVersion controls whether or not a cohabitating resource attempts to serialize into the same
+	// groupVersion as all other cohabitators.  If not, it uses the groupVersion assigned to itself instead of the one being
+	// used by the first enabled cohabitator.
+	ignoreCohabitatingStorageVersion bool
```

Comment:
> If we want that, why would we do this rather than just making two separate resources that both happen to use the same etcd key?

I think that this structure better describes the intent.  These resources are cohabiting, they just aren't stored in the same format yet.  When we try to summarize this information under a discovery endpoint, I think it will be easier to harvest that information if we're able to inspect cohabiters instead inspecting the different resultant etcd locations and prefixes from across the config.